### ISSUE TITLE
Upgrading using docker compose doesn't migrate database

### ISSUE THREAD

**Issue by pkoziol** [firefly-iii/firefly-iii#1999]
**Bug description**
I am running Firefly III version 4.7.9 upgraded just a moment ago from 4.7.8.

After upgrading Firefly to the latest version database hasn't been upgraded, so some parts of application didn't work.

**Steps to reproduce**
1. Update version in docker-compose.yml file (I don't use `latest` tag)
2. Run `docker-compose up -d` (automatically pulls new image since tag changed)
    Which in my case failed because of timeout, but container has been started.
    ```
    Status: Downloaded newer image for jc5x/firefly-iii:release-4.7.9
    fireflyiii_db_1 is up-to-date
    Recreating fireflyiii_app_1 ...
    Recreating fireflyiii_app_1
    
    ERROR: for fireflyiii_app_1  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
    
    ERROR: for app  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
    ERROR: An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
    If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).
    ```
3. Open Firefly III - dashboard sould work.
4. Try to create new transaction - Be right back.

**Expected behavior**
There should not be "Be right back".

**Workaround**
Manually migrate database
```
docker-compose exec app php artisan migrate --seed
docker-compose exec app php artisan firefly:upgrade-database
docker-compose exec app php artisan firefly:verify
```
which asks for confirmation when running in production which couldn't be done when running from container startup script.

**Logs**
`docker logs fireflyiii_app_1`

```
Discovered Package: adldap2/adldap2-laravel
Discovered Package: danhunsaker/laravel-flysystem-others
Discovered Package: davejamesmiller/laravel-breadcrumbs
Discovered Package: fideloper/proxy
Discovered Package: laravel/nexmo-notification-channel
Discovered Package: laravel/passport
Discovered Package: laravel/slack-notification-channel
Discovered Package: laravelcollective/html
Discovered Package: nesbot/carbon
Discovered Package: pragmarx/google2fa-laravel
Discovered Package: rcrowe/twigbridge
Package manifest generated successfully.
**************************************
*     Application In Production!     *
**************************************

Command Cancelled!
Updating currency information..
Transfer #1097 ("Selling EUR") has been updated to use EUR instead of PLN.
Transfer #1165 ("Selling EUR") has been updated to use EUR instead of PLN.
Done updating currency information..
Budget limit #7 (part of budget "Lunch") now has a currency setting (Polski Złoty).
Budget limit #8 (part of budget "Lunch") now has a currency setting (Polski Złoty).
Firefly III database is up to date.
User #1 (...koziolek...) has account #174 ("...") which has no transactions.
Amount integrity OK for user #2
Amount integrity OK for user #1
No orphaned transactions!
Error: Journal #1168 has zero transactions. Open table "transaction_journals" and delete the entry with id #1168
All link types OK!
All access tokens OK!
Amount integrity OK!
[2019-01-12 16:12:54] production.ERROR: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field list' (SQL: update `transaction_currencies` set `enabled` = 1, `updated_at` = 2019-01-12 16:12:54 where `id` in (9, 0, 1, 2, 54) and `transaction_currencies`.`deleted_at` is null) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field list' (SQL: update `transaction_currencies` set `enabled` = 1, `updated_at` = 2019-01-12 16:12:54 where `id` in (9, 0, 1, 2, 54) and `transaction_currencies`.`deleted_at` is null) at /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php:664, Doctrine\\DBAL\\Driver\\PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field list' at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:63, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field list' at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:61)
[stacktrace]
#0 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(624): Illuminate\\Database\\Connection->runQueryCallback('update `transac...', Array, Object(Closure))
#1 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(490): Illuminate\\Database\\Connection->run('update `transac...', Array, Object(Closure))
#2 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(423): Illuminate\\Database\\Connection->affectingStatement('update `transac...', Array)
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2646): Illuminate\\Database\\Connection->update('update `transac...', Array)
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(815): Illuminate\\Database\\Query\\Builder->update(Array)
#5 /var/www/firefly-iii/app/Console/Commands/VerifyDatabase.php(193): Illuminate\\Database\\Eloquent\\Builder->update(Array)
#6 /var/www/firefly-iii/app/Console/Commands/VerifyDatabase.php(102): FireflyIII\\Console\\Commands\\VerifyDatabase->enableCurrencies()
#7 [internal function]: FireflyIII\\Console\\Commands\\VerifyDatabase->handle()
#8 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#9 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#10 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#11 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(572): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#12 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call(Array)
#13 /var/www/firefly-iii/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#14 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#15 /var/www/firefly-iii/vendor/symfony/console/Application.php(901): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#16 /var/www/firefly-iii/vendor/symfony/console/Application.php(262): Symfony\\Component\\Console\\Application->doRunCommand(Object(FireflyIII\\Console\\Commands\\VerifyDatabase), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#17 /var/www/firefly-iii/vendor/symfony/console/Application.php(145): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#18 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Application.php(89): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#19 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(122): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#20 /var/www/firefly-iii/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#21 {main}
"}

In Connection.php line 664:

  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field
  list' (SQL: update `transaction_currencies` set `enabled` = 1, `updated_at`
   = 2019-01-12 16:12:54 where `id` in (9, 0, 1, 2, 54) and `transaction_curr
  encies`.`deleted_at` is null)


In PDOConnection.php line 63:

  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field
  list'


In PDOConnection.php line 61:

  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'enabled' in 'field
  list'


Encryption keys generated successfully.
[2019-01-12 16:12:56] production.INFO: OAuth client generated but no user associated.
Personal access client created successfully.
Client ID: 10
Client Secret: ...
[2019-01-12 16:12:56] production.INFO: OAuth client generated but no user associated.
Password grant client created successfully.
Client ID: 11
Client Secret: ...
Application cache cleared!
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v4.7.9!                                |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
...typical stuff I think...
```

Workaround:
```
koziolek@server:~/docker/firefly-iii$ docker-compose exec app php artisan migrate --seed
**************************************
*     Application In Production!     *
**************************************

 Do you really wish to run this command? (yes/no) [no]:
 > yes

Migrating: 2018_11_06_172532_changes_for_v479
Migrated:  2018_11_06_172532_changes_for_v479
Seeding: AccountTypeSeeder
[2019-01-12 16:23:28] production.INFO: Could not create account type "Default account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Cash account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Asset account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Expense account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Revenue account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Initial balance account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Beneficiary account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Import account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Loan". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Reconciliation account". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Debt". It might exist already.
[2019-01-12 16:23:28] production.INFO: Could not create account type "Mortgage". It might exist already.
Seeding: TransactionCurrencySeeder
[2019-01-12 16:23:28] production.INFO: Could not create transaction currency "EUR". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction currency "PLN". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction currency "USD". It might exist already.
Seeding: TransactionTypeSeeder
[2019-01-12 16:23:29] production.INFO: Could not create transaction type "Withdrawal". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction type "Deposit". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction type "Transfer". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction type "Opening balance". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create transaction type "Reconciliation". It might exist already.
Seeding: PermissionSeeder
[2019-01-12 16:23:29] production.INFO: Could not create role "Site Owner". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create role "Demo User". It might exist already.
Seeding: LinkTypeSeeder
[2019-01-12 16:23:29] production.INFO: Could not create link type "Related". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create link type "Refund". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create link type "Paid". It might exist already.
[2019-01-12 16:23:29] production.INFO: Could not create link type "Reimbursement". It might exist already.
Seeding: ConfigSeeder
[2019-01-12 16:23:29] production.WARNING: Database entry exists. Update to latest version (6)
Database seeding completed successfully.
koziolek@server:~/docker/firefly-iii$ docker-compose exec app php artisan firefly:upgrade-database
Updating currency information..
Done updating currency information..
Firefly III database is up to date.
koziolek@server:~/docker/firefly-iii$ docker-compose exec app php artisan firefly:verify
User #1 (...koziolek...) has account #174 ("...") which has no transactions.
Amount integrity OK for user #2
Amount integrity OK for user #1
No orphaned transactions!
Error: Journal #1168 has zero transactions. Open table "transaction_journals" and delete the entry with id #1168
All link types OK!
All access tokens OK!
Amount integrity OK!
```

**Comment by pkoziol** [comment]
Yes, it's set to `production`.

I think my `docker-compose.yml` is based on my non-docker `.env` which also had `APP_ENV=production`, because it was configured around August/September 2017, which was before you fixed #925.

Not a bug then. Thanks!

**Comment by JC5** [comment]
Well I could add the “force” parameter but I am a bit reluctant because it could also destroy stuff. I’m glad you got it fixed without changing much.
